src/opnsense/mvc/app/models/OPNsense/Base/Constraints/UniqueConstraint.php

```diff
@@ -70,9 +70,13 @@ public function validate($validator, $attribute): bool
                 // collect (additional) key fields
                 $keyFields = array($nodeName);
                 $keyFields = array_unique(array_merge($keyFields, $this->getOptionValueList('addFields')));
+                $exclusions = array_unique($this->getOptionValueList('exclusions'));
```

Comment:
@swhite2 It might be an option to change this to a regex, I think that solves this one https://github.com/opnsense/core/issues/6775 assigned to @fichtner too

Reply:
This is a sort of "allowEmpty" but for a different value.... I'm dealing with a small uphill battle with an exception mask for the nodeValue... we can disable the unqiue constraint, but that doesn't make sense... we need to add more fields (in my case the interface field) if we encounter a special value... otherwise I'm able to create the same link-local address twice to the same interface. I'll finish this on Monday.

Reply:
When combinations of fields should exclude a specific pattern, it might be better to fix the validation in the model itself in stead of using the constraint (not generic enough). In which case my comment might not be relevant for this field type. (I think you can add fields, but combining with a regen has a high chance of getting obscure in that case)

Reply:
Yeah, I'll think a bit more about it. The constraint would make more sense as it uses less code in the long run, but if the options are unreadable that doesn't make it any better.